electron/src/electron-main.js

```diff
@@ -0,0 +1,292 @@
+'use strict'
+
+const { app, Menu, BrowserWindow, shell, session, dialog } = require('electron');
+const childProcess = require('child_process');
+const path = require('path')
+const url = require('url');
+
+// This adds refresh and devtools console keybindings
+// Page can refresh with cmd+r, ctrl+r, F5
+// Devtools can be toggled with cmd+alt+i, ctrl+shift+i, F12
+require('electron-debug')({enabled: true, showDevTools: false});
+require('electron-context-menu')({});
+
+global.eval = function() { throw new Error('bad!!'); }
+
+app.commandLine.appendSwitch('ssl-version-fallback-min', 'tls1.2');
+app.commandLine.appendSwitch('--no-proxy-server');
+app.setAsDefaultProtocolClient('skycoin');
+
+// Keep a global reference of the window object, if you don't, the window will
+// be closed automatically when the JavaScript object is garbage collected.
+let win;
+
+var server = null;
+const serverPort= 8412;
+
+// It is only possible to make connections to hosts that are in this list.
+var allowedHosts = new Map();
+allowedHosts.set('127.0.0.1:' + serverPort, true);
+allowedHosts.set('api.coinmarketcap.com', true);
+allowedHosts.set('api.github.com', true);
+
+// Indicates if the URLs of the nodes were already added to allowedHosts.
+var nodesAllowed = false;
+
+function startServer() {
+  console.log('Starting the local server');
+
+  if (server) {
+    console.log('Server already running');
+    return
+  }
+
+  // Resolve server binary location
+  var exePath = path.dirname(app.getPath('exe'));
+  var exe = (() => {
+    switch (process.platform) {
+      case 'darwin':
+        return path.join(exePath, './../server')
+      case 'win32':
+        return path.join(exePath, './server.exe');
+      case 'linux':
+        return path.join(exePath, './server');
+      default:
+        return path.join(exePath, './server.exe');
+    }
+  })()
+
+  var contentsPath = (() => {
+    switch (process.platform) {
+      case 'darwin':
+        return path.join(exePath, './../dist/')
+      case 'win32':
+        return path.join(exePath, './dist/');
+      case 'linux':
+        return path.join(exePath, './dist/');
+      default:
+        return path.join(exePath, './dist/');
+    }
+  })()
+
+  // Start the server
+  server = childProcess.spawn(exe, ['-port=' + serverPort, '-path=' + contentsPath]);
+
+  createWindow();
+
+  server.on('error', (e) => {
```

Comment:
I always end up adding `e` into `console.log` when debugging, it would be nice to have it there, also with the exit codes..

Reply:
I added `e ` to `console.log`, but I'm not sure if that's exactly what you had in mind